hnn_core/network_builder.py

```diff
@@ -575,10 +642,47 @@ def _parnet_connect(self):
                         nc_dict['A_weight'] = drive_conn[
                             receptor]['A_weight']
                         self._connect_celltypes(
-                            drive['name'], target_cell_type,
+                            connectivity,
                             drive_conn['location'], receptor, nc_dict,
                             unique=drive['cell_specific'])
 
+    def _all_to_all_connect(self, src_types, src_gids, target_types,
+                            target_gids, allow_autapses=True,
+                            unique=False):
+        """Generate connectivity list given lists of sources and targets.
+
+        Parameters
+        ----------
+        src_types : list of str
+            List of string names of sources.
+        src_gids : list on int
+            List of integer identifiers of sources.
+        target_types : list of str
+            List of string names of targets.
+        target_gids : list on int
+            List of integer identifiers of targets.
+        allow_autapses : bool
+            If True, allow connecting neuron to itself.
+        unique : bool
+            If True, each target cell gets one "unique" feed.
+            If False, all src_type cells are connected to
+            all target_type cells.
+        """
+        connectivity = []
```

Comment:
I see that you use {} and [] for initializing a dict and list. It's a bit non-pythonic IMO to not use the name of the object, plus from experience we've seen that sometimes folks confuse between {} and [], so better to just say dict() and list()